src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php

```diff
@@ -67,7 +68,7 @@ public function loadClassMetadata(ClassMetadata $metadata)
                                 $metadata->addAttributeGroup(lcfirst($matches[2]), $group);
                             }
                         } else {
-                            throw new \BadMethodCallException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
+                            throw new MappingException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
```

Comment:
This looks more like a LogicException than a RuntimeException (which MappingException extends from). But MappingException is indeed more consistent with the other loaders. But LoaderInterface doesn't mention what exception can be thrown.